Plugins/SwiftProtobufPlugin/plugin.swift

```diff
@@ -189,15 +228,101 @@ struct SwiftProtobufPlugin: BuildToolPlugin {
             outputFiles: outputFiles
         )
     }
+    
+    
+    /// Matches files from the given path and parent directory. If the given path is a file, it will return the file immediately.
+    /// - Parameters:
+    ///   - path: The file path that you want to search. Paths should be terminated with `/`.
+    ///   - parent: The parent directory of the path.
+    /// - Returns: An array of proto files from the directory and all its subdirectories.
+    private func getProtoFiles(
+        from path: String,
+        in parent: String
+    ) throws -> [String] {
+        let url = URL(fileURLWithPath: parent.appending("/\(path)"))
+        var files = [String]()
+        if url.hasDirectoryPath {
+            if let enumerator = FileManager.default.enumerator(
```

Comment:
This part here is where it gets a bit tricky. SPM does have an internal path and file management implementation that works across different OS's. I would rather we add globbing support to SPM than to reinvent the wheel here.